aclmapping/bank/mapping_test.go

```diff
@@ -0,0 +1,88 @@
+package aclbankmapping
+
+import (
+	"testing"
+
+	pp "github.com/k0kubun/pp/v3"
+	"github.com/stretchr/testify/require"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+
+	"github.com/cosmos/cosmos-sdk/crypto/keys/secp256k1"
+	"github.com/cosmos/cosmos-sdk/simapp"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	"github.com/cosmos/cosmos-sdk/x/bank"
+	"github.com/cosmos/cosmos-sdk/x/bank/types"
+)
+
+
+func cacheTxContext(ctx sdk.Context) (sdk.Context, sdk.CacheMultiStore) {
+	ms := ctx.MultiStore()
+	// TODO: https://github.com/cosmos/cosmos-sdk/issues/2824
+	msCache := ms.CacheMultiStore()
+	return ctx.WithMultiStore(msCache), msCache
+}
+
+
+func TestSendToModuleAccount(t *testing.T) {
```

Comment:
If this approach works then we won't need to really do validation at runtime as we can enforce it un the unit tests.
 
We can also add another one just for the ante handlers

Reply:
One cavet that I haven't figured out is how the prefix store works, I tried adding a GetPrefix() method to the prefix store and all the other stores associated with it but in the CacheKv Store it's always returning empty prefixes